foundations/azure/referenceImplementations/core/managementGroupTemplates/policyDefinitions/Compliant-CorpLzPolicySetDefinition.json

```diff
@@ -10,7 +10,133 @@
+                            },
+                            "privateDnsZoneId": {
+                                "type": "String",
+                                "metadata": {
+                                    "displayName": "Configure a private DNS Zone ID for sqlServer groupID",
+                                    "description": "Configure private DNS zone group to override the DNS resolution for a sqlServer groupID private endpoint.",
+                                    "strongType": "Microsoft.Network/privateDnsZones",
+                                    "assignPermissions": true
+                                }
+                            }
+                        },
+                        "policyRule": {
+                            "if": {
+                                "allOf": [
+                                    {
+                                        "field": "type",
+                                        "equals": "Microsoft.Network/privateEndpoints"
+                                    },
+                                    {
+                                        "field": "Microsoft.Network/privateEndpoints/privateLinkServiceConnections[*].privateLinkServiceId",
+                                        "contains": "Microsoft.Sql/servers"
+                                    },
+                                    {
+                                        "count": {
+                                            "field": "Microsoft.Network/privateEndpoints/privateLinkServiceConnections[*].groupIds[*]",
+                                            "where": {
+                                                "field": "Microsoft.Network/privateEndpoints/privateLinkServiceConnections[*].groupIds[*]",
+                                                "equals": "sqlServer"
+                                            }
+                                        },
+                                        "greaterOrEquals": 1
+                                    }
+                                ]
+                            },
+                            "then": {
+                                "effect": "[[parameters('effect')]",
+                                "details": {
+                                    "type": "Microsoft.Network/privateEndpoints/privateDnsZoneGroups",
+                                    "roleDefinitionIds": [
+                                        "/providers/Microsoft.Authorization/roleDefinitions/4d97b98b-1d4f-4787-a291-c67834d212e7"
+                                    ],
+                                    "existenceCondition": {
+                                        "allOf": [
+                                            {
+                                                "field": "Microsoft.Network/privateEndpoints/privateDnsZoneGroups/privateDnsZoneConfigs[*].privateDnsZoneId",
+                                                "equals": "[parameters('privateDnsZoneId')]"
```

Comment:
Please escape the ARM template functions properly using this pattern "[[parameter('paramName')]" if not, it is not deployable.